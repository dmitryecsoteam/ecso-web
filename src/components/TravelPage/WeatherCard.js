import React from 'react';
import moment from 'moment';

export default ({ tempMin, tempMax, conditionText, conditionImage, date }) => {
    moment.locale('en');
    return (
        <div className="weather-card__container">
            <div className="weather-card__icon">
                <img src={conditionImage}/>    
            </div>
            <div className="weather-card__text-container">
                <span className="weather-card__title">Average weather on <b>{moment(date).format('DD MMMM YYYY')}</b></span>
                <div>
                    <div className="weather-card__temperature">{tempMin}°...{tempMax}°</div>
                    <div className="weather-card__condition">{conditionText}</div>
                </div>
            </div>


        </div>
    )
}