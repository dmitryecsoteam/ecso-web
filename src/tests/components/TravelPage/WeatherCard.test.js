import React from 'react';
import { shallow } from 'enzyme';
import WeatherCard from '../../../components/TravelPage/WeatherCard';
//import moment from 'moment';
jest.unmock('moment');

test('should render WeatherCard with cloud condition', () => {
    const wrapper = shallow(<WeatherCard tempMin={15} tempMax={23} condition="cloud" date="2019-06-01" />);
    expect(wrapper).toMatchSnapshot();
});

test('should render WeatherCard with sun condition', () => {
    const wrapper = shallow(<WeatherCard tempMin={15} tempMax={23} condition="sun" date="2019-06-01" />);
    expect(wrapper).toMatchSnapshot();
});

test('should render WeatherCard with sun-cloud condition', () => {
    const wrapper = shallow(<WeatherCard tempMin={15} tempMax={23} condition="sun-cloud" date="2019-06-01" />);
    expect(wrapper).toMatchSnapshot();
});

test('should render WeatherCard with rain condition', () => {
    const wrapper = shallow(<WeatherCard tempMin={15} tempMax={23} condition="rain" date="2019-06-01" />);
    expect(wrapper).toMatchSnapshot();
});