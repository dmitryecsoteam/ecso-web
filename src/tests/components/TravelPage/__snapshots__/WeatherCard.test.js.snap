// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render WeatherCard 1`] = `
<div
  className="weather-card__container"
>
  <div
    className="weather-card__icon"
  >
    <img
      src="cloud"
    />
  </div>
  <div
    className="weather-card__text-container"
  >
    <span
      className="weather-card__title"
    >
      Average weather on 
      <b>
        01 June 2019
      </b>
    </span>
    <div>
      <div
        className="weather-card__temperature"
      >
        15
        °...
        23
        °
      </div>
      <div
        className="weather-card__condition"
      >
        Mostly cloudy
      </div>
    </div>
  </div>
</div>
`;
